mitmproxy/io/tnetstring.py

```diff
@@ -63,7 +66,14 @@ def dump(value: TSerializable, file_handle: typing.IO[bytes]) -> None:
     This function dumps a python object as a tnetstring and
     writes it to the given file.
     """
-    file_handle.write(dumps(value))
+    try:
+        file_handle.write(dumps(value))
+    except OSError as e:
+        if e.errno == errno.ENOSPC:
+            ctx.log.error("Exiting due to insufficient space on disk")
+            sys.exit(1)
+        else:
+            raise e
```

Comment:
The logic here looks good, but this should really only happen when continuously streaming flows to disk. 

Imagine you have an interactive mitmproxy session, you want to save your flows, there is not enough disk space, and now everything is closed automatically. That would be painful. :) Long story short, let's move this into https://github.com/mitmproxy/mitmproxy/blob/main/mitmproxy/addons/save.py.

Reply:
When running mitmproxy,  if a user decides to stream to a file, it will still exit without a warning because the function executed will be same for both mitmproxy and mitmdump. I think I need to look for a way to detect if the 'save stream to file' was requested by mitmproxy or mitmdump.

Reply:
Yes, the save addon handles that for both. :)